autoPyTorch/pipeline/components/setup/network_backbone/ResNetBackbone.py

```diff
@@ -310,48 +320,51 @@ def _build_block(self, in_features: int, out_features: int) -> nn.Module:
 
     def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
 
-        if self.config["use_skip_connection"]:
-            residual = x
-
         # if shortcut is not none we need a layer such that x matches the output dimension
         if self.shortcut is not None and self.start_norm is not None:
             # in this case self.start_norm is also != none
             # apply start_norm to x in order to have batchnorm+activation
             # in front of shortcut and layers. Note that in this case layers
             # does not start with batchnorm+activation but with the first linear layer
             # (see _build_block). As a result if in_features == out_features
-            # -> result = x + W(~D(A(BN(W(A(BN(x))))))
+            # -> result = x + W_2(~D(A(BN(W_1(A(BN(x))))))
             # if in_features != out_features
             # -> result = W_shortcut(A(BN(x))) + W_2(~D(A(BN(W_1(A(BN(x))))))
             x = self.start_norm(x)
-            if self.config["use_skip_connection"]:
-                residual = self.shortcut(x)
-
-        # TODO make the below code better
-        if self.config["use_skip_connection"]:
-            if self.config["multi_branch_choice"] == 'shake-shake':
-                x1 = self.layers(x)
-                x2 = self.shake_shake_layers(x)
-                alpha, beta = shake_get_alpha_beta(self.training, x.is_cuda)
-                x = shake_shake(x1, x2, alpha, beta)
+            residual = self.shortcut(x)
+        else:
+            if not self.config["use_skip_connection"]:
+                # Early-return
+                return self.layers(x)
             else:
```

Comment:
I think this order (reversed) is easier to understand.
```suggestion
        elif self.config["use_skip_connection"]:
            # We use a skip connection but we do not need to match dimensions
            residual = x
        else:  # Early-return because no need of skip connection
            return self.layers(x)
```